### PR TITLE
feat(generator) : Generator and super-components V1 

### PR DESCRIPTION
**Pull request by aboungnaseng-ovhcloud**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` / `release/**` / `develop` 
| Bug fix?         | yes/no
| New feature?     | yes/no
| Breaking change? | yes/no
| Tickets          | Fix #... 
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

MANAGER-12087: fix bootstrap version 6.4.2 
MANAGER-11573: fix pim error merge
MANAGER-11442: add tracking on ods components
MANAGER-11700: error management in generator
MANAGER-11757: add guide link helpers
MANAGER-11620: msc-billing-tile super-component and super-components utils
MANAGER-11674: msc-advices

### CODE REVIEW

File: packages/manager/core/application/src/ovh-tracking.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        return element;
+      }
+      return false;
+    });
+    return result ? result[0] : '0';
+  };
+
+  // Use trackingPage to send action load
+  const OvhTrackPage = (loc: any) => {
+    const path = loc?.pathname.split('/')[1];
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-tracking.tsx:39]: **suggestion:**
```suggestion
    const [ ,path] = loc?.pathname.split('/');
```

File: packages/manager/core/application/src/ovh-tracking.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    const partsReplace = loc.pathname.substring(1).replaceAll('/', '::');
+    env.then((response) => {
+      const { applicationName, universe } = response;
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-tracking.tsx:42]:
```suggestion
    env.then(({ applicationName, universe }) => {
```

File: packages/manager/core/application/src/ovh-tracking.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+  useEffect(() => {
+    // When location change, send tracking load page
+    OvhTrackPage(location);
+    myStateRef.current = location;
+    setLocationPath(location);
+  }, [location]);
+
+  const ovhTrackingSendCLick = (value: string) => {
+    const path = myStateRef.current.pathname.split('/')[1];
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-tracking.tsx:65]:
```suggestion
    const [ ,path] = myStateRef.current.pathname.split('/');
```

File: packages/manager/core/application/src/ovh-tracking.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    env.then((response) => {
+      const { applicationName, universe } = response;
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-tracking.tsx:67]:
```suggestion
    env.then(({ applicationName, universe }) => {
```

File: packages/manager/core/application/src/ovh-tracking.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+      const name = `${applicationName}::${path || 'homepage'}::${value}`;
+      const page = `${universe}::${applicationName}::${path || 'homepage'}`;
+      tracking.trackClick({
+        name,
+        page: { name: page },
+        complete_page_name: page,
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-tracking.tsx:73]: **question:** Are you sure `trackClick` expects `page` and `complete_page` keys?
I dont see such keys on https://www.npmjs.com/package/ng-at-internet/v/3.1.1

File: packages/manager/core/application/src/ovh-tracking.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+  // Send tracking click for the ODS_COMPONENTS_GROUP
+  const ovhTrackingParentNode = (event) => {
+    const value = event.detail.value || event.detail.newValue;
+    ovhTrackingSendCLick(value);
+  };
+
+  // Init event listener for each ODS_COMPONENTS_GROUP on value change
+  const addOdsEventListenerValue = () => {
+    const elmTab = [];
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-tracking.tsx:88]:
```suggestion
    const elmentTab = [];
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/core/application/src/ovh-tracking.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  };
+
+  useEffect(() => {
+    // Add event listener click on the window
+    window.addEventListener('click', ohvTrackingAction, false);
+    return () => {
+      // clean all events listener
+      window.removeEventListener('click', ohvTrackingAction, false);
+    };
+  }, []);
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-tracking.tsx:143]: **question:** Scanning the entire window object on every user click can cause performance issues no?

File: packages/manager/core/generator/app/conditional-translations/onboarding/Messages_fr_FR.json.hbs
Comment (anooparveti) [inline packages/manager/core/generator/app/conditional-translations/onboarding/Messages_fr_FR.json.hbs:1]: **nitpick:** guideLinks should not be part of translations files

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/core/application/src/ovh-tracking.tsx (hunk 5 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/core/application/src/ovh-tracking.tsx:73]: Its a new request from tracking team

File: packages/manager/core/generator/app/templates/src/components/GuideLink/guide-utils.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    PL: `${docUrl}pl`,
+    PT: `${docUrl}pt`,
+    FR: `${docUrl}fr`,
+    GB: `${docUrl}gb/en`,
+  },
+  CA: {
+    CA: `${docUrl}ca/en`,
+    QC: `${docUrl}ca/fr`,
+    WE: `${docUrl}us/en`,
+    WS: `${docUrl}us/en`,
```
Comment (anooparveti) [inline packages/manager/core/generator/app/templates/src/components/GuideLink/guide-utils.tsx:28]: **nitpick:** Missing Asia, Singapour and India links

File: packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+  return (
+    <>
+      <div className="py-4">
+        <OsdsText
+          level={ODS_TEXT_LEVEL.heading}
+          color={ODS_THEME_COLOR_INTENT.text}
+          size={ODS_TEXT_SIZE._600}
+        >
+          {location.pathname.split('/')[2]}
```
Comment (anooparveti) [inline packages/manager/core/generator/app/templates/src/components/layout-helpers/Dashboard/Dashboard.tsx:45]: **nitpick:** This logic is not true always especially if you're on PCI

File: packages/manager/modules/config/src/locale/country-code.enum.ts [hunk 10]
```diff
@@ -8,4 +8,10 @@ export enum CountryCode {
   PT = 'PT',
   IT = 'IT',
   DE = 'DE',
+  IE = 'IE',
+  QC = 'QC',
+  WE = 'WE',
+  WS = 'WS',
```
Comment (anooparveti) [inline packages/manager/modules/config/src/locale/country-code.enum.ts:14]: **nitpick:** Add India
```suggestion
  WS = 'WS',
  IN = 'IN'
```

File: packages/super-components/_common/mock-helpers.ts [hunk 11]
```diff
@@ -4,8 +4,23 @@ import { Handler } from './msw-helpers';
...
+      return apiClient.v2;
+    case 'ws':
+      return apiClient.ws;
+    default:
+      return apiClient.v6;
+  }
+}
+
+export const mockRequests = (handlers: Handler[], apiVersion: string) => {
+  const mock = new MockAdapter(getApiClient(apiVersion));
```
[the remark this answers is not in the document]
Comment (Kranysys) [reply packages/super-components/_common/mock-helpers.ts:14]: Done ✅

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by chipp972 — APPROVED** [review]

**Pull request merged**